### ISSUE TITLE
Has_many association with the i18n scope return wrong records. Key value backend.

### ISSUE THREAD

**Issue by iamdeuterium** [shioyama/mobility#23]
Example:

```ruby
[53] ruby 2.4.1(main)> i = Dictionaries::Industry.find 2
[54] ruby 2.4.1(main)> i.categories
=> [#<Dictionaries::Category:0x007f44acaae4a0 id: 13, industry_id: 2, slug: "uplotnitelnaya-tehnika">,
 #<Dictionaries::Category:0x007f44acaaddc0 id: 16, industry_id: 2, slug: "dorozhnye-frezy">]
[55] ruby 2.4.1(main)> i.categories.i18n
=> [#<Dictionaries::Category:0x005560c2095a88 id: 13, industry_id: 2, slug: "uplotnitelnaya-tehnika">,
 #<Dictionaries::Category:0x005560c20952e0 id: 16, industry_id: 2, slug: "dorozhnye-frezy">]
[56] ruby 2.4.1(main)> 
[57] ruby 2.4.1(main)> 
[58] ruby 2.4.1(main)> i = Dictionaries::Industry.find 5
[59] ruby 2.4.1(main)> i.categories
=> [#<Dictionaries::Category:0x007f44ac7a5880 id: 36, industry_id: 5, slug: "cazhalki">,
 #<Dictionaries::Category:0x007f44ac7a56a0 id: 37, industry_id: 5, slug: "zhatki">,
 #<Dictionaries::Category:0x007f44ac79aea8 id: 55, industry_id: 5, slug: "uborochnaya-tehnika-dlya-sena-furazha">]
[60] ruby 2.4.1(main)> i.categories.i18n
=> [#<Dictionaries::Category:0x005560c2095a88 id: 13, industry_id: 2, slug: "uplotnitelnaya-tehnika">,
 #<Dictionaries::Category:0x005560c20952e0 id: 16, industry_id: 2, slug: "dorozhnye-frezy">]
[61] ruby 2.4.1(main)> 

```

**Comment by shioyama** [comment]
Thanks, I think I know what's going wrong here, it's independent of what backend you're using.

I've just created a branch called [`do_not_memoize_scope`](https://github.com/shioyama/mobility/tree/do_not_memoize_scope), could you try using that and see if the issue disappears? If so I can fix and release a new version.

```ruby
gem 'mobility', github: 'shioyama/mobility', branch: 'do_not_memoize_scope'
```

**Comment by shioyama** [comment]
Some specs are failing on that branch, related to the `where.not` pattern, so it may not be a complete fix. But please try it.

I'll look deeper into this issue on the weekend (maybe earlier, but it may be tricky...)

**Comment by shioyama** [comment]
Ok specs are now passing, although I'm not sure I totally understand the issue yet. Please try that branch, if it fixes the problem for you I will try to write a spec that tests this issue and then we can release a new version with a fix.

Thanks for reporting!

**Comment by iamdeuterium** [comment]
Now it works correctly. :)

**Comment by shioyama** [comment]
Great thanks. I'll have a new release out today with the fix.

**Comment by shioyama** [comment]
Released 0.1.13 with the fix. Let me know if you notice any other strange things like this one :smile:
